drivers/display/ls0xx.c

```diff
@@ -0,0 +1,441 @@
+}
+
+static void *ls0xx_get_framebuffer(const struct device *dev)
+{
+	LOG_ERR("not supported");
+	return NULL;
+}
+
+static int ls0xx_set_brightness(const struct device *dev,
+				const uint8_t brightness)
+{
+	LOG_WRN("not supported");
+	return -ENOTSUP;
+}
+
+static int ls0xx_set_contrast(const struct device *dev, uint8_t contrast)
+{
+	LOG_WRN("not supported");
+	return -ENOTSUP;
+}
+
+/**
+ * @brief Get display capabilities
+ *
+ * @details Get display capabilities
+ *
+ * @param dev  Pointer to the display device
+ * @param caps Pointer to display_capabilities struct
+ *             this struct will be populated with the
+ *             capabilities
+ */
+static void ls0xx_get_capabilities(const struct device *dev,
+				   struct display_capabilities *caps)
+{
+	memset(caps, 0, sizeof(struct display_capabilities));
+	caps->x_resolution = LS0XX_PANEL_WIDTH;
+	caps->y_resolution = LS0XX_PANEL_HEIGHT;
+	caps->supported_pixel_formats = PIXEL_FORMAT_MONO01;
+	caps->current_pixel_format = PIXEL_FORMAT_MONO01;
+	caps->screen_info = SCREEN_INFO_MONO_MSB_FIRST
+			    | SCREEN_INFO_ALIGNMENT_RESTRICTED;
+	/* Minimum update is of a line */
+	caps->x_alignment = LS0XX_PANEL_WIDTH - 1;
+	caps->y_alignment = 0;
+}
+
+static int ls0xx_set_orientation(const struct device *dev,
+				 const enum display_orientation orientation)
+{
+	LOG_ERR("Unsupported");
+	return -ENOTSUP;
+}
+
+/**
+ * @brief Set pixel format
+ *
+ * @details Set displays pixel format.
+ *          This display is black on white
+ *
+ * @param dev Pointer to the display device
+ * @param pf  Pixel format to set display to
+ *            Only PIXEL_FORMAT_MONO01 is
+ *            supported
+ */
+static int ls0xx_set_pixel_format(const struct device *dev,
+				  const enum display_pixel_format pf)
+{
+	if (pf == PIXEL_FORMAT_MONO01) {
+		return 0;
+	}
+
+	LOG_ERR("not supported");
+	return -ENOTSUP;
+}
+
+/**
+ * @brief Initialize the display
+ *
+ * @details Initialize the display
+ *          Configure the SPI interface options
+ *          Configure GPIO pins for SCS
+ *          Configure DISP and EXTCOMIN pins if defined
+ *          Set DISP pin to OUTPUT_HIGH if defined
+ *          Clear the display and framebuffer
+ *          Write all white pixels to the display
+ *          else it shows random data
+ *
+ * @param dev Pointer to the display device
+ */
+static int ls0xx_init(const struct device *dev)
+{
+
+	struct ls0xx_data *driver = dev->data;
+
+	driver->spi_dev = device_get_binding(DT_INST_BUS_LABEL(0));
+	if (driver->spi_dev == NULL) {
+		LOG_ERR("Could not get SPI device for LS0XX");
+		return -EIO;
+	}
+
+	driver->spi_config.frequency = DT_INST_PROP(0, spi_max_frequency);
+	driver->spi_config.operation = SPI_OP_MODE_MASTER | SPI_WORD_SET(8);
+	driver->spi_config.slave = 0;
+	driver->spi_config.operation |= SPI_TRANSFER_LSB;
+	driver->spi_config.operation |= SPI_CS_ACTIVE_HIGH | SPI_HOLD_ON_CS | SPI_LOCK_ON;
+
+	driver->cs_ctrl.gpio_dev = device_get_binding(
```

Comment:
CS over gpio should be optional, else it will conflict with spi controller dedicated CS line, if any is available. see https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/ethernet/eth_enc424j600.c#L616 as example.

Reply:
@jfischer-no 
Which pin is to be used if user does not mention `cs-gpios`? Is it left to the user in application code to make CS line active?

Reply:
Many SPI controllers have dedicated CS line, few not and uses GPIO for that, which is a bit slower. https://github.com/zephyrproject-rtos/zephyr/blob/master/include/drivers/spi.h#L124 , https://docs.zephyrproject.org/latest/reference/devicetree/api.html#spi